src/npc.h

```diff
@@ -460,6 +460,11 @@ class npc : public player
 
         void load_npc_template( const string_id<npc_template> &ident );
 
+        /** This method is not valid to invoke on a NPC. **/
```

Comment:
Then you should probably add an assertion or a `debugmsg` to report that abuse.

Instead of adding the shared pointer to the `player` class, why not add it to the `game` class? Make it a const value and initialize it with the pointer from `game::u_ptr` (which is initialized once when the `pimpl` is created and is never changed after that):

```
game::game() : u_ptr(), u( *u_ptr ), u_shared_ptr( &u, []( player* ) {} ), ...
```

This avoids changing the *public* interface of the player class.